### ISSUE TITLE
Forms errors are not associated to their field with ARIA

### ISSUE THREAD

**Issue by juliemoynat** [CalderaWP/Caldera-Forms#1500]
Hi,

I was looking for a plugin that build accessible forms and choose Caldera. But, I found an issue with error messages that are not associated to their field with ARIA. So, screen readers can't read them when the user focused a field with error.

Here is the code:
```
<div role="field" data-field-wrapper="fld_6009157" class="form-group has-error">
	<label id="fld_6009157Label" for="fld_6009157_1" class="control-label">Email address <span aria-hidden="true" role="presentation" class="field_required" style="color:#ee0000;">*</span></label>
	<div class="">
		<input required="" data-field="fld_6009157" class="form-control parsley-error" id="fld_6009157_1" name="fld_6009157" value="" data-type="email" aria-required="true" aria-labelledby="fld_6009157Label" data-parsley-id="13" type="email">			
	</div>
	<span class="help-block caldera_ajax_error_block filled" id="parsley-id-13"><span class="parsley-type">This value should be a valid email.</span></span>
</div>
```

1. The `.help-block` element has an ID. This ID should be used to create a link with the field thanks to the `aria-describedby` attribute.
2. The text in the `.help-block` element can change so, it should have an `aria-live="polite"` attribute. Screen readers would read the text as soon as it's changing.
3. The field should have an `aria-invalid="true"` attribute to indicate it has an error.
4. The error message should be in a `<p>` tag has a paragraph.

Here is an example of code with corrections:

```
<div role="field" data-field-wrapper="fld_6009157" class="form-group has-error">
	<label id="fld_6009157Label" for="fld_6009157_1" class="control-label">Email address <span aria-hidden="true" role="presentation" class="field_required" style="color:#ee0000;">*</span></label>
	<div class="">
		<input required="" data-field="fld_6009157" class="form-control parsley-error" id="fld_6009157_1" name="fld_6009157" value="" data-type="email" aria-required="true" aria-labelledby="fld_6009157Label" data-parsley-id="13" type="email" aria-describedby="parsley-id-13" aria-invalid="true">			
	</div>
	<p class="help-block caldera_ajax_error_block filled" id="parsley-id-13" aria-live="polite"><span class="parsley-type">This value should be a valid email.</span></p>
</div>
```

Are these corrections could be done?

**Comment by Shelob9** [comment]
@juliemoynat Thanks for point out this bug. I will get this fixed in the next release. 

The tricky part here is that the element #parsely-id-13 is dynamically created by Parsely (the front-end validation library we use). 13 is effectively a random number.

So, I think solution is when the error element is added to the DOM to do something like `$( '#fld_6009157_1' ).attr( 'aria-describedby', 'parsely-id-13' ).attr( 'aria-invalid', 'true );` Does that make sense.

**Comment by Shelob9** [comment]
Also wondering if solution is submit PR to Parsely... Can do both.

**Comment by Shelob9** [comment]
Related with Sad Face: https://github.com/guillaumepotier/Parsley.js/issues/640

**Comment by juliemoynat** [comment]
Ah yes, sorry!

**Comment by Shelob9** [comment]
Should be fixed via f99286d

@juliemoynat can you please pull master branch and see if that fixes your bug?

Also, see: https://github.com/CalderaWP/Caldera-Forms/blob/master/assets/js/parsley-aria.js

**Comment by juliemoynat** [comment]
Thank you @Shelob9! I will try it tomorrow and give you a feedback.

**Comment by juliemoynat** [comment]
Hi @Shelob9  

I tested with NVDA 2016.4 + Firefox and Jaws 18 + Firefox. It's correct for me.

`span`s are not `p`s but that's already a really good point to have a link between error message and field with error.

When does the next release would be done?

**Comment by Shelob9** [comment]
@juliemoynat Thanks for testing and thanks again for pointing out this bug.

I can't say for sure when this release will be out, but probably in 1-2 weeks.

**Comment by 123aqua** [comment]
@Shelob9  Hi can you help me with something When I fill up phone number it says undefined I'm using caldera forms  here is the screen shot
![16](https://user-images.githubusercontent.com/59171353/72647437-8ca87180-39b3-11ea-9d2c-986827f3f4f8.png)
